pkgs/stdenv/generic/setup.sh

```diff
@@ -1121,7 +1121,13 @@ substituteAllInPlace() {
 # the environment used for building.
 dumpVars() {
     if [ "${noDumpEnvVars:-0}" != 1 ]; then
-        install -m 0600 <(export 2>/dev/null) "$NIX_BUILD_TOP/env-vars" || true
+        # On darwin, install(1) cannot be called with /dev/stdin or fd from process substitution
+        # so first we create the file and then write to it
+        # See https://github.com/NixOS/nixpkgs/issues/335016
+        {
+            install -m 0600 /dev/null "$NIX_BUILD_TOP/env-vars" && \
+            export 2>/dev/null >| "$NIX_BUILD_TOP/env-vars"
+        } || true
```

Comment:
I'm still unclear on why we want to `|| true` this; the whole point of this issue and PR is that "this is a problem." It certainly would be nice to pick up on problems here sooner (i.e. in CI prior to merge into master / unstable) due to a failure rather than later. If we didn't `|| true`, we might not be dealing with this issue in the first place. Just my 2¢.

Reply:
It's not in the scope of this PR or the previous one to change that. It was al ready there, and it was almost 9 years ago: 774f74b8757d6c9809825d3c14fbe313db1378cf.

Reply:
It looks like now that `$NIX_BUILD_TOP` isn’t shared among builds(!), the motivation for the `|| true` no longer applies?

Reply:
My point is that we are getting sidetracked. Can we just get this merged and discuss the implications of failing the command or not in a different issue?

The `|| true` was already there so nothing is changed in that regard.

Reply:
> It's not critical if it fails, thus it can be ignored.

If the functionality is truly so unimportant as to not be worth "fixing right," should it just be removed? That would accomplish the goal of silencing the complaint and decrease maintenance burden.

But if there is room between being "critical" at best and being "unworthy of keeping in the codebase" at worst, it seems reasonable to make sure it works. It does not seem very "nix" to be satisfied with "yeah it works sometimes, and when it doesn't we just ignore it."

That said, I agree it is not in the "scope" of this issue to address it, so I won't bring it up again here.

Reply:
Sure, let’s fix this as it is and then do a follow‐up PR.